File: nattybear/level2/124-나라의-숫자/solution.js
function solution(n) {
  const xs = []
  while (true) {
    if (n == 0) {
      break
    }
    if (n < 3) {
      xs.unshift(n)
      break
    }
    let rem = n % 3
    let quo = Math.floor(n / 3)
    if (rem == 0) {
      quo = quo - 1
      rem = 3
    }
    xs.unshift(rem)
    n = quo
  }
  return xs.join('').replace(/3/g, '4')
}

File: nattybear/level2/124-나라의-숫자/Solution.java
class Solution {
  public String solution(int n) {
    StringBuilder sb = new StringBuilder();
    while (true) {
      if (n == 0)
        break;
      if (n < 3) {
        sb.insert(0, n);
        break;
      }
      int rem = n % 3;
      int quo = n / 3;
      if (rem == 0) {
        quo = quo - 1;
        rem = 3;
      }
      sb.insert(0, rem);
      n = quo;
    }
    return sb.toString().replace('3', '4');
  }
}
